src/syntax/ast.rs

```diff
@@ -1533,7 +1533,13 @@ impl Closure {
     ///
     /// This only exists if you use the function syntax sugar: `let f(x) = y`.
     pub fn name(&self) -> Option<Ident> {
-        self.0.children().next()?.cast()
+        match self.0.cast_first_match::<Pattern>() {
```

Comment:
could this use the ? operator?

Reply:
I updated this, but it feels like there is still some room for improvement, I'm just not sure how.